system_test_bed/test/test_node_crashes.py

```diff
@@ -0,0 +1,81 @@
+#!/usr/bin/env python
+'''
+ * Software License Agreement (BSD License)
+ *
+ *  Copyright (c) 2020, QinetiQ, Inc.
+ *  All rights reserved.
+ *
+ *  Redistribution and use in source and binary forms, with or without
+ *  modification, are permitted provided that the following conditions
+ *  are met:
+ *
+ *   * Redistributions of source code must retain the above copyright
+ *     notice, this list of conditions and the following disclaimer.
+ *   * Redistributions in binary form must reproduce the above
+ *     copyright notice, this list of conditions and the following
+ *     disclaimer in the documentation and/or other materials provided
+ *     with the distribution.
+ *   * Neither the name of QinetiQ nor the names of its
+ *     contributors may be used to endorse or promote products derived
+ *     from this software without specific prior written permission.
+ *
+ *  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ *  "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ *  LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ *  FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ *  COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ *  INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ *  BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ *  LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ *  CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ *  LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ *  ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *  POSSIBILITY OF SUCH DAMAGE.
+'''
+
+import unittest
+import rostest
+import rospy
+import os
+import sys
+from std_msgs.msg import String
+from health_monitor.msg import ReportFault
+
+
+class TestNodeCrashes(unittest.TestCase):
+
+    fault_error = 0
+    fault_count = 0
+    fault_values = {
+        "payload_manager": ReportFault.PAYLOAD_NODE_DIED,
+        "vectornav": ReportFault.AHRS_NODE_DIED,
+        "pressure_sensor": ReportFault.PRESSURE_NODE_DIED,
+        "mission_manager": ReportFault.MISSION_NODE_DIED,
+        "pose_estimator": ReportFault.POSE_NODE_DIED,
+        "thruster_control_node": ReportFault.THRUSTER_NODE_DIED,
+        "autopilot": ReportFault.AUTOPILOT_NODE_DIED,
+        "battery_monitor": ReportFault.BATTERY_NODE_DIED,
+        "jaus_node_bridge": ReportFault.JAUS_NODE_DIED
+    }
```

Comment:
@patapufate consider getting both node name and fault name from arguments instead.